From acb26def02a40f2a832a66bdbbf3d07742623d08 Mon Sep 17 00:00:00 2001
From: Qubot <1445788683@qq.com>
Date: Thu, 29 Jan 2026 11:00:08 +0800
Subject: [PATCH 11/12] rtk_base: fix MAC address assignment logic to ensure
 unique MACs for all LAN ports and correct port mapping

---
 rtk_openwrt/package/realtek/rtk_base/src/port.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/rtk_openwrt/package/realtek/rtk_base/src/port.c b/rtk_openwrt/package/realtek/rtk_base/src/port.c
index 60ecfa6e..9f65c072 100755
--- a/rtk_openwrt/package/realtek/rtk_base/src/port.c
+++ b/rtk_openwrt/package/realtek/rtk_base/src/port.c
@@ -222,11 +222,15 @@ int setup_port(void)
 		{
 			int port;
 			set_interfac_state(phyIf, 0);
-			set_interfac_mac(phyIf, mac, i + 2);  // LAN ports: offset 2, 3, 4, ...
+			set_interfac_mac(phyIf, mac, i + 1);  // LAN ports: offset 1, 2, 3, ... (eth0.2=1, eth0.3=2)
 			//set_interfac_state(phyIf, 1);
-			port = _port_get_phy_port(i);
-			if (port >= 0) {
-				_port_set_dev_port_mapping(port,phyIf);
+			
+			// Only do port mapping for configured LAN ports (within SW_PORT_NUM)
+			if(i < SW_PORT_NUM) {
+				port = _port_get_phy_port(i);
+				if (port >= 0) {
+					_port_set_dev_port_mapping(port, phyIf);
+				}
 			}
 		}
 	}
-- 
2.34.1

